packages/ts/generator-typescript-plugin-model/src/ModelSchemaProcessor.ts

```diff
@@ -0,0 +1,200 @@
+  isComposedSchema,
+  NonComposedSchema,
+  isReferenceSchema,
+  isNumberSchema,
+  isIntegerSchema,
+  isStringSchema,
+  isMapSchema,
+  isArraySchema,
+  isBooleanSchema,
+  convertReferenceSchemaToPath,
+  convertReferenceSchemaToSpecifier,
+  ReferenceSchema,
+  ArraySchema,
+  MapSchema,
+  NonComposedRegularSchema,
+  isNonComposedRegularSchema,
+} from '@vaadin/generator-typescript-core/Schema.js';
+import PluginError from '@vaadin/generator-typescript-utils/PluginError.js';
+import type DependencyManager from '@vaadin/generator-typescript-utils/dependencies/DependencyManager';
+import type { ArrayLiteralExpression, Expression, Identifier, TypeNode } from 'typescript';
+import ts from 'typescript';
+
+type AnnotatedSchema = NonComposedRegularSchema & Readonly<{ 'x-annotations': ReadonlyArray<string> }>;
+
+function isAnnotatedSchema(schema: Schema): schema is AnnotatedSchema {
+  return isNonComposedRegularSchema(schema) && 'x-annotations' in schema;
+}
+
+type AnnotationPrimitiveArgument = boolean | number | string;
+type AnnotationArgument = AnnotationPrimitiveArgument | AnnotationNamedArguments;
+interface AnnotationNamedArguments {
+  readonly [name: string]: AnnotationArgument;
+}
+
+interface Annotation {
+  simpleName: string;
+  arguments: ReadonlyArray<AnnotationArgument>;
+}
+
+function parseAnnotation(annotationText: string): Annotation {
+  const [, simpleName, argumentsText] = /^(\w+)\((.*)\)$/.exec(annotationText) || [];
+  if (simpleName === undefined) {
+    throw new PluginError(`Unknown annotation format when processing "${annotationText}"`);
+  }
+  let args: ReadonlyArray<AnnotationArgument> = [];
+  if (argumentsText !== undefined) {
+    const argumentsTextWithQuotedKeys = argumentsText.replace(/(\w+)\s?:/g, '"$1":');
+    args = JSON.parse(`[${argumentsTextWithQuotedKeys}]`);
+  }
+  return { simpleName, arguments: args };
+}
+
+function convertAnnotationArgumentToExpression(arg: AnnotationArgument): Expression {
+  switch (typeof arg) {
+    case 'boolean':
+      return arg ? ts.factory.createTrue() : ts.factory.createFalse();
+    case 'number':
+      return ts.factory.createNumericLiteral(arg);
+    case 'string':
+      return ts.factory.createStringLiteral(arg);
+    case 'object':
+      return ts.factory.createObjectLiteralExpression(
+        Object.entries(arg).map(([key, value]) =>
+          ts.factory.createPropertyAssignment(key, convertAnnotationArgumentToExpression(value)),
+        ),
+        false,
+      );
+    default:
+      return ts.factory.createOmittedExpression();
+  }
+}
+
+export default class ModelSchemaProcessor {
+  readonly #schema: Schema;
+  readonly #dependencies: DependencyManager;
+  readonly #cwd: string;
+
+  public constructor(schema: Schema, dependencies: DependencyManager, cwd: string) {
+    this.#schema = schema;
+    this.#dependencies = dependencies;
+    this.#cwd = cwd;
+  }
+
+  public process(): [TypeNode, TypeNode, Identifier, ArrayLiteralExpression] {
```

Comment:
Since it is a tuple that shouldn't be changed, let's make it `readonly` and name each parameter. Probably it would be better to extract it to a type.